🐛 Bug: Cannot find module '~~~/ts-api-utils/lib/tokens' from comments.js

### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/template-typescript-node-package/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

`import * as tsutils from "ts-api-utils"` should not throw an error.

### Actual

```plaintext
node:internal/errors:477
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/josh/repos/ts-api-utils/lib/tokens' imported from /Users/josh/repos/ts-api-utils/lib/comments.js
```

### Additional Info

Right now, this repo's source code needs to write `import` statements with explicit `.js` extensions for relative file paths:

```ts
import { greet } from "./greet.js";
```

It does so because the output ESM (ECMAScript Modules) `.js` files need that explicit `.js` extension. The ESM import algorithm doesn't allow omitting it. You can read more at https://www.typescriptlang.org/docs/handbook/esm-node.html.

But, I broke this in https://github.com/JoshuaKGoldberg/ts-api-utils/commit/97690be6798edba9b099c2aad763fb7558a3db8a 😱 

```ts
import { forEachToken } from "./tokens";
```

--- a/src/comments.ts
+++ b/src/comments.ts
@@ -3,7 +3,7 @@
 
 import * as ts from "typescript";
 
-import { forEachToken } from "./tokens";
+import { forEachToken } from "./tokens.js";
 
 /** Exclude trailing positions that would lead to scanning for trivia inside JsxText */
 function canHaveTrailingTrivia(token: ts.Node): boolean {

--- a/src/nodes/utilities.ts
+++ b/src/nodes/utilities.ts
@@ -7,7 +7,7 @@ import {
 	isConstAssertionExpression,
 	isEntityNameExpression,
 	isNumericOrStringLikeLiteral,
-} from "./typeGuards";
+} from "./typeGuards.js";
 
 /** Determines whether a call to `Object.defineProperty` is statically analyzable. */
 export function isBindableObjectDefinePropertyCall(

--- a/src/types/getters.ts
+++ b/src/types/getters.ts
@@ -7,7 +7,7 @@ import {
 	isIntersectionType,
 	isUnionType,
 	isUniqueESSymbolType,
-} from "./typeGuards";
+} from "./typeGuards.js";
 
 export function getCallSignaturesOfType(
 	type: ts.Type

--- a/src/types/utilities.ts
+++ b/src/types/utilities.ts
@@ -9,20 +9,20 @@ import {
 	isObjectFlagSet,
 	isSymbolFlagSet,
 	isTypeFlagSet,
-} from "../flags";
+} from "../flags.js";
 import {
 	isBindableObjectDefinePropertyCall,
 	isInConstContext,
-} from "../nodes/utilities";
-import { isNumericPropertyName } from "../syntax";
-import { getPropertyOfType } from "./getters";
+} from "../nodes/utilities.js";
+import { isNumericPropertyName } from "../syntax.js";
+import { getPropertyOfType } from "./getters.js";
 import {
 	isIntersectionType,
 	isLiteralType,
 	isObjectType,
 	isTupleTypeReference,
 	isUnionType,
-} from "./typeGuards";
+} from "./typeGuards.js";
 
 /** Determines whether the given type is a boolean literal type and matches the given boolean literal. */
 export function isBooleanLiteralType(type: ts.Type, literal: boolean): boolean {

